src/bun.js/api/bun/socket.zig

```diff
@@ -943,10 +942,10 @@ fn NewSocket(comptime ssl: bool) type {
             });
 
             if (!result.isEmptyOrUndefinedOrNull() and result.isAnyError(handlers.globalObject)) {
+                this.detached = true;
+                defer this.markInactive();
```

Comment:
Little worried about use after free here

Reply:
assuming I understand `defer` correctly, this should be run right before any of those `return`s below, which judging from existing use of this and the function calls below in other places seem... okay? :sweat_smile:

Reply:
IIRC, defer runs after return. Like try {return foo(); } catch {} finally {} in JS